web-nfc.d.ts

```diff
@@ -0,0 +1,90 @@
+// Type definitions for Web NFC
+// Project: https://github.com/w3c/web-nfc
+// Definitions by: Takefumi Yoshii <https://github.com/takefumi-yoshii>
+// TypeScript Version: 3.9
+
+// This type definitions referenced to WebIDL.
+// https://w3c.github.io/web-nfc/#actual-idl-index
+
+interface Window {
+  NDEFMessage: NDEFMessage
+}
+declare class NDEFMessage {
+  constructor(messageInit: NDEFMessageInit)
+  records: ReadonlyArray<NDEFRecord>
+}
+declare interface NDEFMessageInit {
+  records: NDEFRecordInit[]
+}
+
+declare type NDEFRecordDataSource = string | BufferSource | NDEFMessageInit
+
+interface Window {
+  NDEFRecord: NDEFRecord
+}
+declare class NDEFRecord {
+  constructor(recordInit: NDEFRecordInit)
+  readonly recordType: string
+  readonly mediaType?: string
+  readonly id?: string
+  readonly data?: DataView
+  readonly encoding?: string
+  readonly lang?: string
+  toRecords?: () => NDEFRecord[]
+}
+declare interface NDEFRecordInit {
+  recordType: string
+  mediaType?: string
+  id?: string
+  encoding?: string
+  lang?: string
+  data?: NDEFRecordDataSource
```

Comment:
Those are not optional in the `dictionary NDEFRecordInit`.

Reply:
```typescript
const myRecord = new NDEFRecord({ recordType: 'text' })
// Error! Property 'data' is missing in type '{ recordType: string; }'
// but required in type 'NDEFRecordInit'.
```

Above compile error is correct? If so, I should fix with another PR.